pkg/providers/vsphere/internal/templates/factory.go

```diff
@@ -112,12 +113,17 @@ func (f *Factory) importOVAIfMissing(ctx context.Context, templateName, ovaURL s
 		return fmt.Errorf("failed to validate template in library for new template: %v", err)
 	}
 
-	if !templateExistsInLibrary {
-		logger.V(2).Info("Importing template from ova url", "ova", ovaURL)
-		if err = f.client.ImportTemplate(ctx, f.templateLibrary, ovaURL, templateName); err != nil {
-			return fmt.Errorf("failed importing template into library: %v", err)
+	if templateExistsInLibrary {
```

Comment:
I'm probably not following here but are we always removing and reimporting?

Reply:
Yeah if an OVA in the library matches the name of the wanted OVA, this would delete it and re-import. Leftover OVAs from previously failed imports would have size 0 GB, so the alternative would be parsing through the data to get the size of the OVA and only deleting it if the size equals 0. I can look into taking this route instead

Reply:
Yea, checking up on other ways to do this, this is the best way forward in my opinion. If we want to skip ContentLibrary altogether, we are restricted to download the ova first locally and then run the import, which massively increases the time to import the template. 

I prefer the idea of checking for the size before deleting the file.